amaranth-stubs/lib/data.pyi

```diff
@@ -186,15 +186,15 @@ class FlexibleLayout(Layout):
     
 
 
-class View(ValueCastable, Generic[_T_ShapeCastable]):
-    def __init__(self, layout: _T_ShapeCastable, target: ValueLike) -> None:
+class View(ValueCastable, Generic[_T_ShapeLike]):
+    def __init__(self, layout: _T_ShapeLike, target: ValueLike) -> None:
```

Comment:
This is unnecessarily loose. The `layout` argument actually needs to be a layout. This changes type errors into runtime errors.